src/R-Tree_Intersection_Finder_C++.cpp

```diff
@@ -137,21 +135,50 @@ libsupermesh::RTree::RTree(const int &dim, const double *positions,
   : dim(dim), visitor(nelements) {
   this->memory = StorageManager::createNewMemoryStorageManager();
   this->buffer = StorageManager::createNewRandomEvictionsBuffer(*this->memory, capacity, bWriteThrough);
+  
+  // Properties as used in PropertySet version of createAndBulkLoadNewRTree in
+  // src/rtree/RTree.cc, libspatialindex 1.8.5
+  Tools::PropertySet properties;
     
-  // Modified version of code from createAndBulkLoadNewRTree in
-  // src/rtree/RTree.cc in libspatialindex 1.8.5. Added 2016-02-24.     
-  id_type indexIdentifier = 0;  // ??
-  this->tree = SpatialIndex::RTree::createNewRTree(*this->buffer, fillFactor,
-    indexCapacity, leafCapacity, this->dim, SpatialIndex::RTree::RV_RSTAR,
-    indexIdentifier);
-  uint32_t bindex = static_cast<uint32_t>(std::floor(static_cast<double>(indexCapacity * fillFactor)));
-  uint32_t bleaf = static_cast<uint32_t>(std::floor(static_cast<double>(leafCapacity * fillFactor)));
+  Tools::Variant treeVariant;
+  treeVariant.m_varType = Tools::VT_LONG;
+  treeVariant.m_val.lVal = SpatialIndex::RTree::RV_RSTAR; 
+  properties.setProperty("TreeVariant", treeVariant);
+  
+  Tools::Variant v_fillFactor;
+  v_fillFactor.m_varType = Tools::VT_DOUBLE;
+  v_fillFactor.m_val.dblVal = fillFactor;
+  properties.setProperty("FillFactor", v_fillFactor);
+  
+  Tools::Variant v_indexCapacity;
+  v_indexCapacity.m_varType = Tools::VT_ULONG;
+  v_indexCapacity.m_val.ulVal = indexCapacity;
+  properties.setProperty("IndexCapacity", v_indexCapacity);
+  
+  Tools::Variant v_leafCapacity;
+  v_leafCapacity.m_varType = Tools::VT_ULONG;
+  v_leafCapacity.m_val.ulVal = leafCapacity;
+  properties.setProperty("LeafCapacity", v_leafCapacity);
+  
+  Tools::Variant dimension;
+  dimension.m_varType = Tools::VT_ULONG;
+  dimension.m_val.ulVal = dim;
+  properties.setProperty("Dimension", dimension);
+  
+  Tools::Variant pageSize;
+  pageSize.m_varType = Tools::VT_ULONG;
+  pageSize.m_val.ulVal = std::numeric_limits<uint32_t>::max() / 2;
```

Comment:
For reference: this is as large as possible to attempt to avoid disk swapping. This value is later multiplied by the `ExternalSortBufferTotalPages` property, which must be at least 2, hence the divide.

Reply:
I have added a comment to this effect.